Packs/Hackuity/Integrations/Hackuity/Hackuity.py

```diff
@@ -0,0 +1,694 @@
+                },
+                {
+                    "total": 0,
+                    "open": 0,
+                    "closed": 0,
+                    "ignored": 0,
+                },
+            )
+            outputs.append(output)
+        return outputs
+
+
+""" HELPER FUNCTIONS """
+
+
+def json_get_value(item: Any, path: List[Union[str, None]], default: Any = None) -> Any:
+    for key in path:
+        if item is None:
+            item = {}
+        if key is None:  # None means "any key"
+            try:
+                item = next(iter(item.values()))
+            except StopIteration:
+                item = None
+        else:
+            item = item.get(key)
+    if item is None:
+        return default
+    return item
+
+
+def json_set_value(item: Dict[str, Any], path: List[str], value: Any) -> None:
+    if value is None:
+        return
+    leaf = path.pop()
+    for key in path:
+        if key not in item:
+            item[key] = {}
+        item = item[key]
+    item[leaf] = value
+
+
+def json_extract(
+    data: Dict[str, Any],
+    mapping: Dict[str, List[Union[str, None]]],
+    defaults: Dict[str, Any] = {},
+) -> Dict[str, Any]:
+    return {
+        key: json_get_value(data, path, defaults.get(key))
+        for key, path in mapping.items()
+    }
+
+
+def json_format(
+    data: Dict[str, Any],
+    mapping: Dict[str, List[str]],
+    formatters: Dict[str, Callable[[Any], Any]] = {},
+) -> Dict[str, Any]:
+    output: Dict[str, Any] = {}
+    for key, path in mapping.items():
+        value = data[key]
+        if formatter := formatters.get(key):
+            value = formatter(value)
+        json_set_value(output, path, value)
+    return output
+
+
+def format_date(date_str: Optional[str]) -> Optional[str]:
+    """Format a date from API"""
+    if date_str:
+        return datetime.strftime(dateutil.parser.parse(date_str), DATE_FORMAT)
+    return None
+
+
+def none_or_apply(text: Optional[str], apply: Callable[[str], Any]) -> Optional[Any]:
+    if text is None:
+        return None
+    return apply(text)
+
+
+""" COMMAND FUNCTIONS """
+
+
+def test_module(client: Client) -> str:
+    """Tests API connectivity and authentication"""
+    # excption formating in case of authentication error
+    # is already done in client.get_access_token
+    set_integration_context({})  # reset cache
+    client.fetch_user()
+    return "ok"
+
+
+def hackuity_search_findings_command(
+    client: Client, args: Dict[str, Any]
+) -> CommandResults:
+    asset_name = args.get("asset_name")
+    asset_type = args.get("asset_type")
+    attribute = args.get("attribute")
+    limit = int(args["limit"])
+    cvss_min = none_or_apply(args.get("cvss_min"), float)
+    cvss_max = none_or_apply(args.get("cvss_max"), float)
+    vuln_type = args.get("vuln_type")
+    trs_min = none_or_apply(args.get("trs_min"), int)
+    trs_max = none_or_apply(args.get("trs_max"), int)
+    findings = client.fetch_findings(
```

Comment:
make sure in case there's non, a readable output is returned an not an error

Reply:
If there are no findings, `fetch_findings` will return an empty list, I think this is ok

![screenshot-2022-01-26-18:17:26](https://user-images.githubusercontent.com/3994389/151213336-b4c8199d-b714-4139-98e3-b6c3578c6c46.png)

Reply:
sounds good too!